lib/index.js

```diff
@@ -146,7 +147,7 @@ api.openCollections = brCallbackify(async names => {
   await Promise.all(unopened.map(async name => {
     logger.debug('creating collection: ' + name);
     try {
-      await api.db.createCollection(name);
+      await api.db.createCollection(name, {writeConcern});
```

Comment:
Since the `writeConcern` is passed into the client on initialization in `connectionOptions` is this also required?

Reply:
it is actually not needed here, it is needed here though:
https://github.com/digitalbazaar/bedrock-mongodb/blob/31a3c1d8a39a6655735d537382f4a3f68da581d5/lib/index.js#L162-L164

Reply:
Can you confirm that `https://github.com/mongodb/node-mongodb-native/blob/3.6/lib/db.js#L436-L437` doesn't automatically merge in the defaults that were set earlier when the client was made?

Reply:
Similarly, it looks like `db.createCollection` follows this code path to merge in the `writeConcern` from `db`:

https://github.com/mongodb/node-mongodb-native/blob/3.6/lib/db.js#L538

https://github.com/mongodb/node-mongodb-native/blob/3.6/lib/operations/create_collection.js#L27-L29

https://github.com/mongodb/node-mongodb-native/blob/e8ac5585068e8fe897cb302f44a8e64b7fa201cc/lib/operations/command.js#L34

https://github.com/mongodb/node-mongodb-native/blob/e8ac5585068e8fe897cb302f44a8e64b7fa201cc/lib/utils.js#L437

So it seems that adding `writeConcern` in either place shouldn't be needed -- can we confirm that if we don't include it that the options aren't set properly (i.e., that they don't use the options we set on `db`)?

Reply:
can confirm it is needed in the `db.collection` call. Perhaps the code you are referencing is in the next release of the driver?

Reply:
Code is from version 3.6 of the driver:

https://github.com/mongodb/node-mongodb-native/blob/3.6/lib/db.js#L75 
https://github.com/mongodb/node-mongodb-native/blob/3.6/lib/db.js#L186

How are you confirming it's needed?

Reply:
Also, you linked to the function call `db()` which would be called on a client instance which would have been created via this: `http://mongodb.github.io/node-mongodb-native/3.6/api/MongoClient.html` (which takes `writeConcern`).

Reply:
code as is:

```js
Running Mocha tests...

  api
    openCollection
      ✓ should open a collection
      ✓ should open several collections (53ms)
    createIndex
      ✓ should create an index
    collections
      ✓ should have collections
      ✓ should insertOne into a collection
      ✓ should insertMany into a collection
      ✓ should findOne in a collection
      ✓ should find many records in a collection
```

code with:

```js
  await Promise.all(unopened.map(async name => {
    collections[name] = await api.db.collection(name, {writeConcern});
  }));
```

changed to:

```js
  await Promise.all(unopened.map(async name => {
    collections[name] = await api.db.collection(name);
  }));
```

results in:

```js
  api
    openCollection
Top-level use of w, wtimeout, j, and fsync is deprecated. Use writeConcern instead.
      ✓ should open a collection
Top-level use of w, wtimeout, j, and fsync is deprecated. Use writeConcern instead.
Top-level use of w, wtimeout, j, and fsync is deprecated. Use writeConcern instead.
Top-level use of w, wtimeout, j, and fsync is deprecated. Use writeConcern instead.
      ✓ should open several collections (42ms)
    createIndex
      ✓ should create an index
    collections
      ✓ should have collections
      ✓ should insertOne into a collection
      ✓ should insertMany into a collection
      ✓ should findOne in a collection
      ✓ should find many records in a collection


  8 passing (120ms)
```

also in the issue you can see where mongodb (the company/ maintainer of the driver)'s own test suite is throwing these depreaction notices due to defaults in their write api.

Reply:
It looks like `collectionKeys` is missing `writeConcern`:

https://github.com/mongodb/node-mongodb-native/blob/3.6/lib/db.js#L385-L395

Which means that [mergeOptionsAndWriteConcern](https://github.com/mongodb/node-mongodb-native/blob/3.6/lib/db.js#L437) fails to copy that over. We should comment on having to pass `writeConcern` to `db.collection()` on that basis, and we should use the write concern from the `db` instance, not from the config -- effectively "fixing" the bug in the driver.